apps/graphql/src/apps/itinerary/types/outputs/ItineraryInterface.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+
+import {
+  GraphQLInterfaceType,
+  GraphQLString,
+  GraphQLList,
+  GraphQLBoolean,
+} from 'graphql';
+import GlobalID from '@kiwicom/graphql-global-id';
+import differenceInMinutes from 'date-fns/differenceInMinutes';
+
+import type { Sector } from '../../../common/CommonTypes';
+import type { Itinerary } from '../../Itinerary';
+import GraphQLPrice from '../../../common/types/outputs/Price';
+import GraphQLSector from '../../../common/types/outputs/Sector';
+import GraphQLRouteStop from '../../../common/types/outputs/RouteStop';
+
+const getStopoverDuration = (
+  sector: GraphQLSector,
+  previousSector: ?GraphQLSector,
+): number | null => {
+  if (previousSector == null) {
+    return null;
+  }
+
+  const currentSectorDeparture = sector.departure?.time?.utc;
+  const previousSectorArrival = previousSector.arrival?.time?.utc;
+
+  if (currentSectorDeparture == null || previousSectorArrival == null) {
+    return null;
+  }
+
+  return differenceInMinutes(
+    new Date(currentSectorDeparture),
+    new Date(previousSectorArrival),
+  );
+};
+
+export const commonFields = {
+  id: GlobalID(({ id }) => id),
+  price: { type: GraphQLPrice },
+  sectors: {
```

Comment:
We should then refactor this to use just `sector` for `OneWay` and `inbound/outbound` for `Return`. Like in case of booking and https://kiwi-search-schema.now.sh/voyager . But this is definitely for different PR.

Otherwise looks good. Thanks

Reply:
Yes, my ideas was to do it step by step. This PR is big enough as it is

Reply:
Definitely, so lets merge it 😉